components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileProducer.java

```diff
@@ -212,6 +215,28 @@ protected void processExchange(Exchange exchange, String target) throws Exceptio
                 }
             }
 
+            // any checksum file to write?
+            if (endpoint.getChecksumFileAlgorithm() != null) {
+                String algorithm = endpoint.getChecksumFileAlgorithm();
+                String checksumFileName = target + "." + algorithm;
+
+                // create exchange with checksum as body to write as the checksum file
+                MessageHelper.resetStreamCache(exchange.getIn());
+                InputStream is = exchange.getIn().getMandatoryBody(InputStream.class);
+                Exchange checksumExchange = new DefaultExchange(exchange);
+                checksumExchange.getIn().setBody(new DigestUtils(algorithm).digestAsHex(is));
+
+                LOG.trace("Writing checksum file: [{}]", checksumFileName);
+                // delete any existing done file
+                if (operations.existsFile(checksumFileName)) {
+                    if (!operations.deleteFile(checksumFileName)) {
+                        throw new GenericFileOperationFailedException(
+                                "Cannot delete existing checksum file: " + checksumFileName);
+                    }
+                }
+                writeFile(checksumExchange, checksumFileName);
```

Comment:
Maybe move this block to a separate method? We have far too much methods / classes that are far too large and we have been broken them in smaller pieces. This would help this effort.

Reply:
Done